File: clients/python/Faraway.py
#!/usr/bin/python

import socket
import simplejson
import time
import httplib
import urllib
import sys

class Client:
    def __init__(self, Host='127.0.0.1', Port=None, SSL=0):
	if Port == None:
	    if SSL:
		conclass = httplib.HTTPSConnection
		Port = 9096
	    else:
		conclass = httplib.HTTPConnection
		Port = 9095

	hoststr = '%s:%d' % (Host, Port)

	self.conn = conclass(hoststr)

    def __getattr__(self, name):
	return lambda data={}: self.call_remote(name, data)

    def call_remote(self, action, data):
        headers = {
            "Content-type": "application/x-www-form-urlencoded",
            "Accept": "text/plain"
        }

	obj = {
	    'timestamp': int(time.time()),
	    'data': data,
	}

	obj['checksum'] = 1

        self.conn.request("POST", '/' + action, simplejson.dumps(obj), headers)
        res = self.conn.getresponse()

	response = simplejson.loads(res.read())
	return response

File: clients/ruby/Faraway.rb
#!/usr/bin/ruby

require 'net/http'
require 'net/https'
require 'uri'
require 'json'

class Faraway
    def initialize(args)
	@ssl = args[:ssl] || false
	@port = args[:port]

	if @port.nil? then
	    @port = @ssl ? 9096 : 9095
	end

	@host = args[:host] || '127.0.0.1'
    end

    def method_missing(m, *args)
	return call_remote(m, args)
    end

    def call_remote(action, data = {})
	http = Net::HTTP.new(@host, @port)
	http.use_ssl=@ssl

	obj = {
	    'timestamp' => Time.now.to_i,
	    'checksum' => 1,
	    'data' => data,
	}

	headers = {
	    'Content-Type' => 'application/x-www-form-urlencoded'
	}

	res, data = http.post("/#{action}", JSON.JSON(obj), headers)

	return JSON.parse(res.body)
    end
end

